From: Zi Yan <ziy@nvidia.com>
Subject: mm-page_isolation-remove-migratetype-from-move_freepages_block_isolate-fix
Date: Thu, 8 May 2025 15:23:42 -0400

Here is the fixup 2/3 to address Johannes' comments.

1. replace get_pageblock_isolate() with
   get_pageblock_migratetype() == MIGRATE_ISOLATE, a
   get_pageblock_isolate() becomes private in mm/page_isolation.c

Link: https://lkml.kernel.org/r/F05CF961-6104-4C46-94C2-8966F580D8AC@nvidia.com
Signed-off-by: Zi Yan <ziy@nvidia.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_alloc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/page_alloc.c~mm-page_isolation-remove-migratetype-from-move_freepages_block_isolate-fix
+++ a/mm/page_alloc.c
@@ -1975,7 +1975,7 @@ static bool __move_freepages_block_isola
 	int from_mt;
 	int to_mt;
 
-	if (isolate == get_pageblock_isolate(page))
+	if (isolate == (get_pageblock_migratetype(page) == MIGRATE_ISOLATE))
 		return false;
 
 	if (!prep_move_freepages_block(zone, page, &start_pfn, NULL, NULL))
_
